pkg/sql/tablewriter.go

```diff
@@ -794,16 +800,42 @@ func (td *tableDeleter) deleteAllRowsFast(
 			EndKey: tablePrefix.PrefixEnd(),
 		}
 	}
-	log.VEventf(ctx, 2, "DelRange %s - %s", resume.Key, resume.EndKey)
-	td.b.DelRange(resume.Key, resume.EndKey, false /* returnKeys */)
-	td.b.Header.MaxSpanRequestKeys = limit
+	// The most efficient deletion can be done using the ClearRange
+	// method. However, to maintain backwards compatibility, if no GC
+	// deadline is set for the table, we must revert to using
+	// DeleteRange. This is necessary because a legacy DROP or TRUNCATE
+	// would not set the GC deadline, but would still expect the data
+	// to remain historically queryable until the zone's TTL expires.
+	if td.tableDesc().GCDeadline == 0 {
+		log.VEventf(ctx, 2, "backwards-compatible DelRange %s - %s", resume.Key, resume.EndKey)
+		td.b.DelRange(resume.Key, resume.EndKey, false /* returnKeys */)
+		td.b.Header.MaxSpanRequestKeys = limit
+		if _, err := td.finalize(ctx, traceKV); err != nil {
+			return resume, err
+		}
+		if l := len(td.b.Results); l != 1 {
+			panic(fmt.Sprintf("%d results returned", l))
+		}
+		return td.b.Results[0].ResumeSpan, nil
+	}
+
+	// Assert that GC deadline is in the past.
+	if timeutil.Since(timeutil.Unix(0, td.tableDesc().GCDeadline)) < 0 {
+		panic(fmt.Sprintf("not past the GC deadline: %s", td.tableDesc()))
+	}
+	log.VEventf(ctx, 2, "ClearRange %s - %s", resume.Key, resume.EndKey)
```

Comment:
And this needs to be added to `(p *planner) ShowTrace` too.

Reply:
Added to `ShowTrace`. Unfortunately it's really difficult to see either of these because both are run through an asynchronous pathway which doesn't show up `SHOW KV TRACE...`.